### ISSUE TITLE
[Bug][Authz] `CREATE OR REPLACE` operation requires `ALTER` table privilege instead of `CREATE` table privilege when table exists

### ISSUE THREAD

**Issue by wang-zhun** [apache/incubator-kyuubi#3845]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

role A
```
CREATE VIEW default.view_tst(a, b) AS SELECT 1, 2;

DESC default.view_tst;

+---------+---------+----------+
| col_name|data_type|   comment|
+---------+---------+----------+
|        a|      int|      null|
|        b|      int|      null|
+---------+---------+----------+

```

role B
```
CREATE OR REPLACE VIEW default.view_tst(c, d) AS SELECT 1, 2;
```

role A
```
DESC default.view_tst;

+---------+---------+----------+
| col_name|data_type|   comment|
+---------+---------+----------+
|        c|      int|      null|
|        d|      int|      null|
+---------+---------+----------+
```

B has the permission to create tables in the default database, but does not have the permission to modify default.view_tst.
In fact, B can be modified successfully, indicating that there is a permission leak in the table of A

### Affects Version(s)

master

### Are you willing to submit PR?

- [ ] Yes. I can submit a PR independently to fix.
- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

**Comment by pan3793** [comment]
cc @bowenliang123 @zhouyifan279

**Comment by bowenliang123** [comment]
Currently the opType for privilege check, soly relies on the node name of the plan, which lacks of detail of plan.
1.see calling `RuleAuthorization` <https://github.com/apache/incubator-kyuubi/blob/master/extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleAuthorization.scala#L50>
2.see mapping `CreateViewCommand` to `CREATEVIEW` in `OperationType` <https://github.com/apache/incubator-kyuubi/blob/master/extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/OperationType.scala#L73>

In order to satisfy determining the opType of command , we have to,
1. modify `OperationType.apply` to make it accept plan detail
2. use the detail of the plan to determine opType, in this case, we may rely on the `allowExisting` attribute of `CreateViewCommand` to check whether the view exsited.

**Comment by wang-zhun** [comment]
@bowenliang123 `ReplaceTable` and `ReplaceTableAsSelect` also require `r.catalog.tableExists(r.tableName)`

**Comment by bowenliang123** [comment]
The `opType` (as operationType) is now hardcoded string value in spec json as in `CommandSpec`.
We may need to find a way to decouple `opType` with `classname` in the spec, and allow it to be changed in extractor in runtime. 
cc @yaooqinn

**Comment by attilapiros** [comment]
@pan3793 I have run into this problem but instead of checking for an ALTER privilege my suggestion would be checking for two permissions, either for
1) CREATE and DROP
or
2) CREATE and ALTER

My preference is CREATE and DROP but we can discuss this.

I even have a PR ready where I modified `table_command_spec.json`  to store multiple `opTypes`  for a command.

**Comment by pan3793** [comment]
@attilapiros I'm not an expert in this area, have you investigated what permissions other systems(for example, PostgreSQL, Snowflake, Databricks Unity Catalog) check in this case?

**Comment by attilapiros** [comment]
I just checked impala and hive but neither has CREATE OR REPLACE so that does not help. 
I will look around at other DBs at the end of this week.

**Comment by attilapiros** [comment]
# PostgreSQL

 PostgreSQL does not support a CREATE OR REPLACE TABLE syntax. 

# Snowflake

Claude says no extra privileges are needed just CREATE.

# Databricks Unity Catalog

Claude answer for "Can you summarize what privileges are needed in Unity Catalog for a CREATE OR REPLACE TABLE?"

```
....
So the practical answer is: USE CATALOG + USE SCHEMA + CREATE TABLE, plus ownership (or MANAGE) of the existing table — but treat the ownership part as my inference from the drop/alter model, not a verbatim doc quote. I'd confirm it by testing before citing it authoritatively.
```

Unfortunately I cannot validate Unity Catalog easily. But an ownership / MANAGE permission suggest me we need an extra permission (either ALTER or DROP).

**Comment by attilapiros** [comment]
@pan3793 I found a description of the necessary rights at one of the plugin's example:
```
To overwrite an existing table, you must have the following privileges:
- USE CATALOG on the catalog
- USE SCHEMA on the schema
- SELECT on the table
- MODIFY on the table

For more information, see [Unity Catalog privileges reference](https://docs.databricks.com/gcp/en/data-governance/unity-catalog/access-control/privileges-reference).
``` 

Source: https://docs.databricks.com/gcp/en/integrations/excel-write-back#:~:text=There%20is%20no%20separate%20setting%20in%20the,more%20information%2C%20see%20Unity%20Catalog%20privileges%20reference.

**Comment by pan3793** [comment]
@attilapiros, thanks for the investigation. I took a closer look at the Databricks docs, and figured out why `SELECT` and `MODIFY` are more appropriate here.

https://docs.databricks.com/gcp/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using#parameters

> REPLACE
> 
> If specified, replaces the table and its content if it already exists. This clause is only supported > for Delta and Apache Iceberg tables.
> 
> REPLACE preserves the [table history](https://docs.databricks.com/gcp/en/tables/history) and granted privileges, row filters, and column masks.

In OSS Spark, `CREATE OR REPLACE` is not allowed for v1 tables; and for v2 tables, if the data source implements the `StagingTableCatalog`, it calls `StagingTableCatalog#stageReplace` (connector implementation-dependent), otherwise, it performs `DROP` then `CREATE`.

So, we can check `CREATE` and `DROP`/`ALTER`(or something similar to `MODIFY`) based on whether the type connector implements `StagingTableCatalog` if we can distinguish that, otherwise, I would follow Databricks Unity Catalog behavior given that the mainstream lakehouse table formats (iceberg, delta) would implement `StagingTableCatalog`.

**Comment by attilapiros** [comment]
Yes, only Paimon (https://paimon.apache.org/docs/master/spark/sql-ddl) and iceberg is effected.

**Comment by attilapiros** [comment]
I am working on PR

**Comment by attilapiros** [comment]
@pan3793 could you please check this PR and share your oppinion: https://github.com/apache/kyuubi/pull/7748
